waiter/src/waiter/auth/authentication.clj

```diff
@@ -254,7 +255,9 @@
               user (assoc "x-waiter-auth-user" (str user))))))
 
 (defn remove-done-param-from-keep-alive-https-redirect
-  "Removes the done=true query parameter from the response when it is an https redirect to the keep-alive endpoint."
+  "Removes the done=true query parameter from the response when it is an https redirect to the keep-alive endpoint.
+   Keeping the done=true parameter would prevent authentication to be triggered for soon to expire cookies after
+   the requests comes back with https."
```

Comment:
```suggestion
   Keeping the done=true parameter would prevent authentication from being triggered for soon-to-expire cookies after
   the requests comes back with https."
```

Reply:
Is this correct?  I still don't follow entirely.  Why would there be a redirect inside the auth handler to HTTPS that also contains `done=true` as well?